TFNetworkRecLayer.py

```diff
@@ -2135,13 +2178,24 @@ def maybe_transform(layer):
             with tf.name_scope("end_flag"):
               end_flag = cur_end_layer.output.placeholder
               end_flag = tf.logical_or(end_flag, self.net.layers["end"].output.placeholder)  # (batch * beam,)
+
+              # Translate the beams of end_flags to the latest choices in the net.
+              last_choices = self._latest_choice_layer.get_search_choices()
+              resolved_end_layer = last_choices.translate_to_this_search_beam(cur_end_layer)
+              end_flag_resolved = resolved_end_layer.transform_func(
+                cur_end_layer.output.placeholder if rec_layer.include_eos else end_flag)
             with tf.name_scope("dyn_seq_len"):
-              dyn_seq_len = cur_end_layer.transform_func(dyn_seq_len)
+              # Translate beams of dyn_seq_len from previous to current frame (going through all choices)
+              for choice_layer in reversed(self._choice_seq_in_frame):
+                dyn_seq_len = select_src_beams(dyn_seq_len, src_beams=choice_layer.search_choices.src_beams)
```

Comment:
Don't call `select_src_beams` here, and esp do not replicate the logic about unrolling over the choice seq. Just use the existing logic (e.g. call `transform_func`).
In any case, as said, seq len and end flag, both should be resolved in the same way.

Reply:
Yeah, I was aware of this, but the alternative here would be to get the "prev:_latest_choice_layer", transform it to the current frame with `translate_to_this_search_beam` and then never use the output, but only use `transform_function` to transform `dyn_seq_len`. This didn't feel right either... Or is there a better way?